### ISSUE TITLE
[Feature] Ability to Set `GRAPH_ETH_GET_LOGS_MAX_CONTRACTS` Per Provider

### ISSUE THREAD

**Issue by itsjerryokolo** [graphprotocol/graph-node#4974]
### Description

cc @leoyvens 

### Some information to help us out

- [ ] Tick this box if you plan on implementing this feature yourself.
- [X] I have searched the issue tracker to make sure this issue is not a duplicate.

**Comment by azf20** [comment]
Thanks @itsjerryokolo, @leoyvens do you think this should be set at the provider level?

**Comment by leoyvens** [comment]
Yes I think so, the env var would be a default if the provider hasn't set anything
